src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -770,8 +779,11 @@ void RadioSpinel<InterfaceType, ProcessContextType>::HandleValueIs(spinel_prop_k
 
         if (status >= SPINEL_STATUS_RESET__BEGIN && status <= SPINEL_STATUS_RESET__END)
         {
-            // If RCP crashes/resets while radio was enabled, posix app exits.
-            VerifyOrDie(!IsEnabled(), OT_EXIT_RADIO_SPINEL_RESET);
+            if (IsEnabled())
+            {
+                HandleUnexpectedReset();
+                ExitNow(error = OT_ERROR_RCP_RECOVERING);
```

Comment:
```suggestion
                ExitNow(error = OT_ERROR_NONE);
```

Can we hide the error ?

Reply:
Updated.